### DISCUSSION TITLE
Improve documentation

### DISCUSSION THREAD

**Discussion by gwyneplaine** [JedWatson/react-select#4869] [General]
Originally, this was an Issue to keep track of ways we need to improve the documentation, it's been converted to a discussion and will be updated as new items are considered and added. Please feel free to add comments below so that we can better discuss how to improve the documentation.

- [x] Add example for getOptionValue method 
- [x] Add documentation around the expected default properties of options (value: [any], label: string).
- [x] Add Async Select section to the upgrade-guide
- [x] #3037 
- [ ] Fix focus styling of buttons to show outline (view source/codesandbox example buttons) for better a11y
- [ ] Add Accessibility section
  - ariaLiveMessages documentation and i18n example
  - An example of setting up a label and connecting to select
  - An inline error state and how to get the screen reader to read it
- [ ] Add examples for debounce with async
- [ ] Add examples for SSR and CSP using NonceProvider
- [ ] Add section regarding use with forms
  - Rendering of hidden fields
  - how it pertains to props `name`, `form`, and `delimiter`
  - Examples with various form managers (ie: react-hook-form, react-final-form, formik, etc...)
- [ ] Support for dark mode https://github.com/JedWatson/react-select/issues/4067

**Comment by danpoq** [comment]
I'm not sure whether this requires its own issue but right now there could be a potential problem with disparity between the contents of the docs site and the npm release.

Got quite confused just now because https://react-select.com/styles#overriding-the-theme is live. I had to go through issue history to find that the PR is merged but there's no npm release.

Seeing as though docs are hosted from `docs/` on `master`. One solution might be to add something like `Since 2.0.1` to each new feature in the docs. Might also be a larger problem with your branch management e.g. `master = latest released stable version`.

**Comment by wejrox** [comment]
The [source links](https://jedwatson.github.io/react-select/) are pointing to the wrong URL for the examples. Presumably they have changed and not been updated.

**Comment by Rall3n** [comment]
@wejrox https://jedwatson.github.io/react-select/ has been replaced with https://v1.react-select.com. There all example links are pointing to the correct source.

**Comment by ebonow** [comment]
- [ ] Add Accessibility section
- [ ] Add examples for debounce with async
- [ ] Add examples for SSR and CSP using NonceProvider
- [ ] Add section regarding use with forms, rendering of hidden fields, how it pertains to props `name`, `form`, and `delimiter` and perhaps examples with various form managers (ie: react-hook-form, react-final-form, formik, etc...)
- [ ] Support for dark mode https://github.com/JedWatson/react-select/issues/4067

**Reply by DelusionCB** [reply]
Any news on the accessibility section? Still trying to figure out how to implement custom i18n messages.

**Comment by moloko** [comment]
Just to point out that there's a whole bunch of stuff - mostly event handlers I think - listed on the [Props page](https://react-select.com/props) that pretty clearly doesn't exist. Certainly it's hard to believe react-select would need to support event handlers like `onTimeUpdate` or `onVolumeChange`...!

**Comment by joeyfigaro** [comment]
I'd like to suggest that the async examples should be fetching from remote sources, since the docs suggest it's what the async select is intended to be used for.

**Reply by moloko** [reply]
You could use [Dog API](https://dog.ceo/dog-api/documentation/) for this...

**Reply by joeyfigaro** [reply]
Great idea @moloko

**Comment by DrogoNevets** [comment]
Any examples on `NonceProvider` yet? we have the CSP issue and im shooting blind on fixing it
